port/linuxs390/omrtime.c

```diff
@@ -230,9 +245,10 @@ omrtime_hires_delta(struct OMRPortLibrary *portLibrary, uint64_t startTime, uint
 	if (OMRTIME_HIRES_CLOCK_FREQUENCY == requiredResolution) {
 		/* no conversion necessary */
 	} else if (OMRTIME_HIRES_CLOCK_FREQUENCY < requiredResolution) {
-		ticks = (uint64_t)((double)ticks * ((double)requiredResolution / (double)OMRTIME_HIRES_CLOCK_FREQUENCY));
+		ticks = muldiv64(ticks, requiredResolution, OMRTIME_HIRES_CLOCK_FREQUENCY);
```

Comment:
Since required resolution has to be one of a few options this code would be modified so that we significantly reduce the cost of calculating the results.  See my comments below about muldiv64 being a much more expensive operation than just using the constants everywhere